plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryUploadAppArtifactTask.kt

```diff
@@ -0,0 +1,124 @@
+package io.sentry.android.gradle.tasks
+
+import io.sentry.android.gradle.extensions.SentryPluginExtension
+import io.sentry.android.gradle.telemetry.SentryTelemetryService
+import io.sentry.android.gradle.telemetry.withSentryTelemetry
+import io.sentry.android.gradle.util.asSentryCliExec
+import javax.inject.Inject
+import org.gradle.api.Project
+import org.gradle.api.file.Directory
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFile
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.work.DisableCachingByDefault
+
+@DisableCachingByDefault(because = "Upload task shouldn't be cached")
+abstract class SentryUploadAppArtifactTask @Inject constructor(objectFactory: ObjectFactory) :
+  SentryCliExecTask() {
+
+  @get:InputFile
+  @get:Optional
+  @get:PathSensitive(PathSensitivity.NAME_ONLY)
+  val bundle: Property<RegularFile> = objectFactory.fileProperty()
+
+  @get:InputDirectory
+  @get:Optional
+  @get:PathSensitive(PathSensitivity.NAME_ONLY)
+  abstract val apk: DirectoryProperty
+
+  override fun getArguments(args: MutableList<String>) {
+    args.add("mobile-app")
+    args.add("upload")
+
+    val bundleFile = bundle.orNull?.asFile
+    if (bundleFile != null) {
+      if (bundleFile.exists()) {
+        args.add(bundleFile.path)
+        return
```

Comment:
actually, if we need to add more arguments to the upload (like proguard or other files), this is perhaps not the best design since it relies on early returns.